backend/internal/server/apis/stats/goals.go

```diff
@@ -0,0 +1,70 @@
+package stats
+
+import (
+	"github.com/gofiber/fiber/v2"
+	"github.com/jak103/powerplay/internal/server/apis"
+	"github.com/jak103/powerplay/internal/server/services/auth"
+	"github.com/jak103/powerplay/internal/utils/responder"
+	"github.com/jak103/powerplay/internal/models"
+	"github.com/jak103/powerplay/internal/db"
+	"github.com/jak103/powerplay/internal/utils/locals"
+)
+
+
+func init() {
+	apis.RegisterHandler(fiber.MethodPost, "/goals/submit", auth.Public, postGoalsHandler)
+}
+
+func postGoalsHandler(c *fiber.Ctx) error {
+	log := locals.Logger(c)
+	log.Info("Handling creating new goal")
+	log.Debug("body: %q", c.Request().Body())
+	goalPostRequest := new(models.Goal)
+	err := c.BodyParser(goalPostRequest)
+	
+	// If valid structure in post request, continue on
+	if err != nil{
+		log.WithErr(err).Error("Failed to parse Goal POST request.")
+		return err
+	}
+
+	// Connect to database and insert goal
+	db := db.GetSession(c)
+	record, err := db.SaveGoal(goalPostRequest)
+	
+	if err != nil{
+		log.WithErr(err).Alert("Failed to parse goal request payload")
+		return responder.InternalServerError(c)
+	}
+
+	if record == nil {
+		return responder.BadRequest(c, "Could not post goal into database")
+	}
+	
+	return responder.Ok(c)
```

Comment:
What do you mean by this? Is the endpoint not working on your end?

Reply:
Would you like me to make an issue for this?

Reply:
We need to figure out how we want to respond after a POST, the two standard ways being either return the data object or just return an OK. I'm just returning an OK on the Penalties API, but if we want to return the object I can change that

Also I figured out using responder.OkWithData() yesterday with Dr. Christensen so that function works

Reply:
Ok sounds good. I will go ahead and fix that.

Reply:
I think this should just be responding with ok. There feels no need to respond with any data (unless I am completely wrong) for this post request.

Reply:
I agree, that's how the Penalty API currently has it